ee/tabby-db/src/lib.rs

```diff
@@ -155,16 +159,40 @@ impl DbConn {
         Self::init_db(pool).await
     }
 
-    /// Initialize database, create tables and insert first token if not exist
-    async fn init_db(pool: SqlitePool) -> Result<Self> {
-        let user_input: String = input(
-            "The database is from an older version. Type 'confirm' to run database migrations:",
-        )
-        .interact()?;
+    fn prompt_confirm() -> Result<bool> {
+        let user_input: String = input("The database is from an older version. Type 'confirm' to run database migrations:")
+            .interact()?;
         if user_input != "confirm" {
             outro("Database initialization cancelled.")?;
             return Err(anyhow!("Database initialization cancelled by user"));
         }
+        Ok(true)
+    }
+
+
+
+    /// Initialize database, create tables and insert first token if not exist
+    async fn init_db(pool: SqlitePool) -> Result<Self> {
+        
+        let local_migrations = sqlx::migrate!();
+        let mut applied_migrations = sqlx::query("SELECT version FROM _sqlx_migrations")
+        .fetch_all(&pool)
+        .await?.into_iter();
```

Comment:
this seems a bit verbose to me - maybe try method like this one https://docs.rs/sqlx/latest/sqlx/struct.SqliteConnection.html#method.list_applied_migrations (as we have a connection to sqlite already)